public/send-results.js

```diff
@@ -0,0 +1,26 @@
+/* global: location, sessionStorage */
+
+(function() {
+  var path = location.pathname.slice(1)
+
+  if (path === "login") {
+    var submitSurveyButton = document.getElementById('submit-survey-button')
+    var clinicianEmailInput = document.getElementById('clinician-email-input')
+    submitSurveyButton.addEventListener('click', function() {
+      var storeClinicianEmail = sessionStorage.setItem('clinician-email', clinicianEmailInput.value)
+    })
+  }
+
+  if (path === "results") {
+    var sendResultsButton = document.getElementById('send-results-button')
+    var getSessionStorage = window.sessionStorage
+    var getClinicianEmail = getSessionStorage['clinician-email']
+    var emailSubject = getSessionStorage['select-survey'] + ' results ' + Date()
+    var emailBody = JSON.stringify(getSessionStorage)
```

Comment:
You could do this.
```js
var storage = window.sessionStorage
var allSessionData = {}
for (var i = 0; i < storage.length; i++) {
  var key = storage.key(i)
  allSessionData[key] = storage.getItem(key)
}
```
But the problem with both that approach and your current approach is that if you want to store other data in you session, its going to pop out unexpectedly.

Right now, the regex you are doing to create the email body is evil. You want to format your json in a nicer way? I expect that escaped characters in the json could do really weird things in the results.

One thing you could do is have a some functions

```js
var saveAnswer = (name, value) => {
  answers = json.parse(sessionStorage.getItem('answers'))
  answers[name] = value
  sessionStorage.setItem('answers', JSON.stringify(answers))
}
```

There are other things you can do also, like maintain an array of keys that you are interested in (I think this is what you have suggested)